src/core/xref.js

```diff
@@ -208,6 +199,28 @@ function getSpecContext(elem) {
   return specs;
 }
 
+/**
+ * @param {HTMLElement} elem
+ * @param {boolean} isIDL
+ */
+function getForContext(elem, isIDL) {
+  if (elem.dataset.xrefFor) {
+    return normalize(elem.dataset.xrefFor);
+  }
+
+  if (!isIDL) {
+    // non-idl terms having for context is not presently supported
+    return null;
+  }
+
+  /** @type {HTMLElement} */
+  const dataXrefForElem = elem.closest("[data-xref-for]");
+  if (dataXrefForElem) {
+    return normalize(dataXrefForElem.dataset.xrefFor);
+  }
```

Comment:
Seems we can save a `return null` if we move this block into `if (isIDL)` 👀

Reply:
Good point.